cmd/allocator/main.go

```diff
@@ -143,7 +141,7 @@ func (h *httpHandler) allocateHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	allocation := h.agonesClient.AllocationV1alpha1().GameServerAllocations(h.namespace)
+	allocation := h.agonesClient.AllocationV1alpha1().GameServerAllocations(agonesSystemNamespace)
```

Comment:
I'm pretty sure you were right the first time - this should be `gsa.ObjectMeta.Namespace` (we could also default this to "default" if wanted to?, since that is the default for everything else) 

What we have here is going to create a GameServerAllocation that (should) searches in the agones-system namespace - because that is the namespace it is coming from. This works right now because the current GSA logic is buggy.

Reply:
Yes. Confusion over confusion ;) for a second I thought gsa is tight to agones-system because it is an API server extension, but it actually receives the namespace that the request is sent to as part of the request. I am defining cluster role to let allocator create gs allocation requests in multiple namespaces.